components/ws-manager/pkg/manager/monitor.go

```diff
@@ -1039,6 +1050,36 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		}
 	}
 
+	var snc wsdaemon.WorkspaceContentServiceClient
+	var wsDaemonIsReachable = func(ctx context.Context, wso workspaceObjects) wait.ConditionWithContextFunc {
+		return func(ctx context.Context) (bool, error) {
+			var conn *grpc.ClientConn
+			snc, conn, err = m.manager.connectToWorkspaceDaemon(ctx, wso)
+			if err != nil {
+				return false, err
```

Comment:
Once we can't connect to ws-daemon, an error returns. So the function `wait.PollImmediateWithContext` would be aborted immediately without retrying within 5 mins, is this expected behavior?

If yes, then the code could be consider removed
```go
		if errors.Is(err, wait.ErrWaitTimeout) {
			msg = "ws-daemon is not running in the node"
		}
```